lib/hooks.js

```diff
@@ -100,8 +100,8 @@ function hookRunnerApplication (hookName, boot, server, cb) {
         exit()
       } else {
         boot(function manageTimeout (err, done) {
-          done(err)
           exit(err)
+          done(err)
```

Comment:
I don't understand. This looks like two separate callbacks being invoked. How is this deterministic?

Reply:
Here we are appending a function to avvio's ready function.
This is needed to catch the timeout exception and bubble the error up to the user.

The done, is the avvio callback and it must be called to continue the loading of the application

The exit function, instead it is the a local function that will append the onready hooks of all the children in the fastify instance tree and, when the is an error it will stop appending the hooks and returns to the user the error
So this function will be executed every time as the last function for every fastify context. 

This has been done to provide to the fastify onready hook a signature without the error handling in charge of the user

Maybe a comment could be helpful?

Reply:
> Maybe a comment could be helpful?

I think so.

Reply:
I agree with @jsumners :)

Reply:
Done